src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+
+    public RightTriangle(String color,double firstLeg, double secondLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    public double calculateArea() {
+        return (firstLeg * secondLeg) * 0.5;
+    }
+
+    @Override
+    public void drawFigure() {
+        System.out.println("Figure: right triangle, area: " + calculateArea()
+                + " sq. units, firstLeg: " + firstLeg + " units, secondLeg: "
+                + secondLeg + " units, color: " + getColor());
```

Comment:
The `drawFigure` method should be overridden from an interface that specifies this behavior, ensuring the class has a single responsibility and adheres to the interface segregation principle.